todo_app/app.py

```diff
@@ -1,48 +1,82 @@
+from flask_login.mixins import AnonymousUserMixin
 from todo_app.data.MongoDbApi import MongoDbApi
 from todo_app.ViewModel import ViewModel
 from todo_app.data.Items import Items
 from flask import Flask, render_template, request, redirect, send_from_directory
 from todo_app.flask_config import Config
 import os
+from flask_login import LoginManager, login_required, current_user
+from todo_app.security.Auth import Auth
 
 
 def create_app(db_name=None):
     app = Flask(__name__)
     app.config.from_object(Config())
     MongoDbApi.init(db_name)
+    Auth.init()
 
     @app.route('/')
+    @login_required
     def index():  # pylint:disable=unused-variable
-        item_view_model = ViewModel(
-            Items.get_items(), Items.get_current_sort_order())
+        item_view_model = None
+        if app.config['LOGIN_DISABLED']:
+            item_view_model = ViewModel(
+                Items.get_items(), Items.get_current_sort_order(), True)
+        else:
+            item_view_model = ViewModel(Items.get_items(
+            ), Items.get_current_sort_order(), current_user.has_write_permission())
```

Comment:
Minor, but from a code cleanliness perspective it's often best to keep the conditional check as small as possible, and this is a nice example of where we can do that;
```suggestion
        can_write = app.config['LOGIN_DISABLED'] or current_user.has_write_permission()
        item_view_model = ViewModel(
                Items.get_items(), Items.get_current_sort_order(), can_write)
        return render_template('index.html', view_model=item_view_model)
```